src/app/features/notification/notification.component.ts

```diff
@@ -35,80 +37,157 @@ import { closeOnlySubmenu } from "./close-only-submenu";
+    this.hasNotificationEnabled = !this.hasNotificationEnabled;
+
+    const notificationConfig = await this.loadNotificationConfig(this.userId);
+
+    // TODO: Currently, email notification are disabled. Update this logic once the email notification feature is implemented.
+    const notificationChannels = {
+      push: this.hasNotificationEnabled,
+      email: false,
+    };
+
+    if (notificationConfig) {
+      notificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(notificationConfig);
+    } else {
+      const newNotificationConfig = new NotificationConfig(this.userId);
+      newNotificationConfig.channels = notificationChannels;
+      await this.entityMapper.save(newNotificationConfig);
+    }
+
+    this.alertService.addInfo(
+      `Notification ${this.hasNotificationEnabled ? "enabled" : "disabled"}.`,
+    );
+  }
+
+  /**
+   * Updates the read status for multiple notifications.
+   */
+  private async updateReadStatusForNotifications(
+    notifications: NotificationEvent[],
+    newStatus: boolean,
+  ) {
+    for (const notification of notifications) {
+      notification.readStatus = newStatus;
+      await this.entityMapper.save(notification);
+    }
+    this.filterUserNotifications(this.allNotifications);
   }
 
+  /**
+   * Updates the read status of a single notification.
+   */
   async updateReadStatus(notification: NotificationEvent, newStatus: boolean) {
     notification.readStatus = newStatus;
     await this.entityMapper.save(notification);
     this.filterUserNotifications(this.allNotifications);
   }
 
-  deleteNotification(notification: NotificationEvent) {
-    // Need to add/update this logic to delete the notification from the CouchDB
-    this.allNotifications = this.allNotifications.filter(
-      (n) => n !== notification,
-    );
-    this.unreadNotifications = this.unreadNotifications.filter(
-      (n) => n !== notification,
-    );
-    Logging.log("Notification deleted");
+  /**
+   * Deletes a user notification.
+   */
+  async deleteNotification(notification: NotificationEvent) {
+    await this.entityMapper.remove(notification);
+    this.alertService.addInfo("Notification deleted successfully");
+    this.loadAndProcessNotifications();
+  }
 
-    this.filterUserNotifications(this.allNotifications);
+  /**
+   * Handles notification events by redirecting the user to the corresponding action URL.
+   * @param {NotificationEvent} notification - The notification event containing the action URL.
+   */
+  notificationListener(notification: NotificationEvent) {
+    this.router.navigate([notification.actionURL]);
   }
 
-  protected readonly closeOnlySubmenu = closeOnlySubmenu;
+  /**
+   * Redirect the user to Notification setting page
+   */
+  onRedirectToNotificationsSetting() {
+    this.router.navigate(["/user-account"], { queryParams: { tabIndex: 1 } });
+  }
```

Comment:
I think you could directly use `[routerLink]="[...]"` directive in the html so that we don't need an extra method in the .ts here

Reply:
(same for notificationListener above?)